### PR TITLE
BUG-92863 fixed resource loading

### PR DESCRIPTION
**Pull request by doktoric**

### CODE REVIEW

File: blueprint-manager/src/main/java/com/sequenceiq/cloudbreak/blueprint/BlueprintSegmentReader.java [hunk 1]
```diff
@@ -50,33 +52,32 @@
...
-                    for (final File configEntry : getFiles(insideFolder)) {
-                        if (configEntry.isFile()) {
-                            String serviceName = serviceEntry.getPath().replace(serviceEntry.getParent() + "/", "");
-                            if (!collectedFiles.keySet().contains(serviceName(serviceName))) {
-                                collectedFiles.put(serviceName(serviceName), templateFiles(Lists.newArrayList()));
-                            }
-                            String file = configEntry.getPath().replace(serviceEntry.getParent() + "/", "");
-                            collectedFiles.get(serviceName(serviceName)).getFiles().add(String.format("%s/%s", dir, file));
+            List<URL> files = getFiles(dir);
+            if (files != null) {
```
Comment (mhmxs) [inline blueprint-manager/src/main/java/com/sequenceiq/cloudbreak/blueprint/BlueprintSegmentReader.java]: `getFiles()` never be null

File: blueprint-manager/src/main/java/com/sequenceiq/cloudbreak/blueprint/BlueprintSegmentReader.java [hunk 2]
```diff
@@ -50,33 +52,32 @@
...
+                for (final URL serviceEntry : files) {
+                    String[] serviceAndPath = serviceEntry.getPath().split(dir);
+                    String insideFolder = String.format("%s/%s", dir, serviceAndPath[1].replaceAll("/", ""));
+                    for (final URL configEntry : getFiles(insideFolder)) {
+                        String serviceName = serviceAndPath[1].replace("/", "");
```
Comment (mhmxs) [inline blueprint-manager/src/main/java/com/sequenceiq/cloudbreak/blueprint/BlueprintSegmentReader.java]: Code duplication

File: cloud-common/src/main/java/com/sequenceiq/cloudbreak/util/FileReaderUtils.java [hunk 3]
```diff
@@ -42,6 +48,19 @@ public static String readFileFromClasspath(String fileName) throws IOException {
...
 
+    public static List<URL> readFolderFromClasspath(String folderName) throws IOException {
+        final ClassLoader loader = Thread.currentThread().getContextClassLoader();
+
+        try (final InputStream is = loader.getResourceAsStream(folderName);
+            final InputStreamReader isr = new InputStreamReader(is, StandardCharsets.UTF_8);
+            final BufferedReader br = new BufferedReader(isr)) {
+            return br.lines()
+                    .map(l -> folderName + "/" + l)
+                    .map(r -> loader.getResource(r))
```
Comment (mhmxs) [inline cloud-common/src/main/java/com/sequenceiq/cloudbreak/util/FileReaderUtils.java]: Can be replaced with method reference

File: cloud-common/src/main/java/com/sequenceiq/cloudbreak/util/FileReaderUtils.java [hunk 4]
```diff
@@ -42,6 +48,23 @@ public static String readFileFromClasspath(String fileName) throws IOException {
...
+            final InputStreamReader isr = new InputStreamReader(is, StandardCharsets.UTF_8);
+            final BufferedReader br = new BufferedReader(isr)) {
+            return br.lines()
+                    .map(l -> getFolderNameAndFileName(folderName, l))
+                    .map(r -> loader.getResource(r))
+                    .collect(toList());
+        }
+    }
+
+    private static String getFolderNameAndFileName(String folderName, String l) {
```
Comment (topolyai5) [inline cloud-common/src/main/java/com/sequenceiq/cloudbreak/util/FileReaderUtils.java]: The name 'l' is not so expressive

File: blueprint-manager/src/main/java/com/sequenceiq/cloudbreak/blueprint/BlueprintSegmentReader.java [hunk 5]
```diff
@@ -50,33 +52,30 @@
...
+            String message = String.format("Could not read files from the definiated folder which was: %s", dir);
+            LOGGER.warn(message, ex);
+            throw new BlueprintProcessingException(message, ex);
         }
 
         return collectedFiles;
     }
 
-    private File[] getFiles(String configDir) throws IOException {
-        File folder = resourceLoader.getResource("classpath:" + configDir).getFile();
```
Comment (topolyai5) [inline blueprint-manager/src/main/java/com/sequenceiq/cloudbreak/blueprint/BlueprintSegmentReader.java]: Maybe could use getInputStream()

File: blueprint-manager/src/main/java/com/sequenceiq/cloudbreak/blueprint/BlueprintSegmentReader.java [hunk 6]
```diff
@@ -50,33 +52,30 @@
...
-                            collectedFiles.get(serviceName(serviceName)).getFiles().add(String.format("%s/%s", dir, file));
-                        }
+            List<URL> files = getFiles(dir);
+            for (final URL serviceEntry : files) {
+                String[] serviceAndPath = serviceEntry.getPath().split(dir);
+                String simpleServiceName = serviceAndPath[1].replaceAll("/", "");
+                String insideFolder = String.format("%s/%s", dir, simpleServiceName);
+                for (final URL configEntry : getFiles(insideFolder)) {
+                    if (!collectedFiles.keySet().contains(serviceName(simpleServiceName))) {
+                        collectedFiles.put(serviceName(simpleServiceName), templateFiles(Lists.newArrayList()));
```
Comment (topolyai5) [inline blueprint-manager/src/main/java/com/sequenceiq/cloudbreak/blueprint/BlueprintSegmentReader.java]: Redundant expression

File: blueprint-manager/src/main/java/com/sequenceiq/cloudbreak/blueprint/BlueprintSegmentReader.java (hunk 5 above)
Comment (doktoric) [reply blueprint-manager/src/main/java/com/sequenceiq/cloudbreak/blueprint/BlueprintSegmentReader.java]: this line was deleted :)

**Pull request merged**